context/MqlContext/MqlContextProvider.tsx

```diff
@@ -64,79 +76,67 @@ function MqlContextProviderInternal({
-    {
-      data: mqlServerUrlData,
-      error: mqlServerUrlError,
-      fetching: mqlServerUrlfetching,
-    },
+    { data: mqlServerUrlData, error: mqlServerUrlError },
     refetchMqlServerUrl,
-  ] = useQuery<MqlServerUrlQuery>({
-    query: gql`
-      query MqlServerUrlQuery {
-        mqlServerUrl
-      }
-    `,
+  ] = useQuery<MqlServerUrlQueryType>({
+    query: MqlServerUrlQuery,
+    pause: !isAuthenticated,
   });
-  if (mqlServerUrlError) throwCarefully(mqlServerUrlError);
+  if (mqlServerUrlError) handleCombinedError(mqlServerUrlError);
 
   const [{ fetching }, setMqlServerUrl] = useMutation<
-    SetMqlServerUrl,
-    SetMqlServerUrlVariables
-  >(
-    gql`
-      mutation SetMqlServerUrl($newServerIdAsString: String!) {
-        createUserPreference(
-          object: {
-            prefKey: "mql_server_override_id"
-            prefValue: $newServerIdAsString
-          }
-          on_conflict: {
-            constraint: user_prefs_user_id_pref_key_key
-            update_columns: prefValue
-          }
-        ) {
-          id
-          prefKey
-          prefValue
-          userId
-          createdAt
-        }
-      }
-    `
-  );
+    SetMqlServerUrlMutationType,
+    SetMqlServerUrlMutationVariables
+  >(SetMqlServerUrlMutation);
 
-  const mqlClient =
-    getToken && mqlServerUrlData?.mqlServerUrl
-      ? buildMqlUrqlClient(getToken, mqlServerUrlData.mqlServerUrl)
-      : null;
-  [];
-
-  const [mqlContext, setMqlContext] = useState<MqlContextType>({
-    setMqlServerUrl: (newServerId: number) =>
+  const setMqlServerUrlThenRefetch = useCallback(
+    (newServerId: number) =>
       // Note: newServerid is cast as a string because it is stored on the backend
       // as a User Preference, whose values are all strings.
       setMqlServerUrl({ newServerIdAsString: `${newServerId}` }).then(
         refetchMqlServerUrl
       ),
+    [setMqlServerUrl, refetchMqlServerUrl]
+  );
+
+  /* 
+    Note: Before the mqlServerUrl is available, an urql client will be constructed with the url set to the CORE_API_URL.
+    This is because it throws an error to build an urql client without a URL.
+    Naturally, this is invalid, so we must check with the mqlServerUrl is present before initiating MQL Queries.
+  */
+  const mqlClient = buildMqlUrqlClient(
+    getToken,
+    mqlServerUrlData?.mqlServerUrl || CORE_API_URL
+  );
+
+  const [mqlContext, setMqlContext] = useState<MqlContextType>({
+    coreApiUrl: CORE_API_URL,
+    mqlServerUrl: mqlServerUrlData?.mqlServerUrl,
+    setMqlServerUrl: setMqlServerUrlThenRefetch,
     mqlServerOverrideLoading: fetching,
+    modelKey: null,
+    setModelKey: () => Promise.resolve(),
+    modalKeyOverrideLoading: false,
     mqlClient,
-    mqlServerUrl: mqlServerUrlData?.mqlServerUrl,
     getToken,
+    useQuery,
+    useMutation,
+    handleCombinedError,
   });
 
   useEffect(() => {
-    const stateToUpdate: MqlContextType = {};
+    const stateToUpdate: any = {};
```

Comment:
Maybe use Partial<MqlContextType>?

It looks like the only other thing we would want to check for changes would be the modelKey once that's in use?  If so, could you add a small comment here to help us (me) remember in the future?

Reply:
Ah smart.